shared/src/ledger/storage/merkle_tree.rs

```diff
@@ -579,11 +567,14 @@ mod test {
         let key_prefix: Key =
             Address::Internal(InternalAddress::PoS).to_db_key().into();
         let pos_key = key_prefix.push(&"test".to_string()).unwrap();
+        let pos_key2 = key_prefix.push(&"test2".to_string()).unwrap();
 
         let ibc_val = [1u8; 8].to_vec();
         tree.update(&ibc_key, ibc_val.clone()).unwrap();
         let pos_val = [2u8; 8].to_vec();
         tree.update(&pos_key, pos_val).unwrap();
+        let pos_val2 = [3u8; 8].to_vec();
+        tree.update(&pos_key2, pos_val2).unwrap();
```

Comment:
Could you add a comment?
```suggestion
        // to prevent the subtree from being empty after deletion
        tree.update(&pos_key2, pos_val2).unwrap();
```